### ISSUE TITLE
Add a rule that PR can be merged only with squashed commits

### ISSUE THREAD

**Issue by SMore-Napi** [InnoQueue/Backend#97]
- Automatically squash commits when merge into `main` branch. No other options are possible
- Make merge button available ONLY after GitHub Actions passed

**Comment by al1ych** [comment]
added rule to squash&merge only in the settings of the repository

**Comment by SMore-Napi** [comment]
If these rules work, add them in Mobile repository

**Comment by al1ych** [comment]
T

> Check all edge cases and if protection is really works - close this issue. Some ideas to check
> 
>     * After new commit the approve should be omitted
> 
>     * If build fails then the merge button is disabled. If GitHub action is fixed in the same PR then merge button should appear.
> 
>     * Only our team can approve PR. Other contributors are not allowed to approve any PR.
> 
>     * Other contributors can not merge even if PR is approved.
> 
>     * ???
> 
>     * Your ideas to check

Wait I'm not done yet :D

**Comment by al1ych** [comment]
Test pull request here: https://github.com/InnoQueue/Backend/pull/122
Let's not delete it just yet

**Comment by al1ych** [comment]
So, about some of the ideas you've mentioned:

- After new commit the approve should be omitted

**This is working fine by default**

- If build fails then the merge button is disabled. If GitHub action is fixed in the same PR then merge button should appear.

**Got it working in this issue (see PR #122)**

- Only our team can approve PR. Other contributors are not allowed to approve any PR.

**I'll look into this in another issue (the smaller the PR/issue, the better separation of concerns)**

- Other contributors can not merge even if PR is approved.

**I think there was something about it in the settings, I'll look into that in the other issue**

**Comment by al1ych** [comment]
The issue is resolved, see the new connected issue for the above-mentioned problems: https://github.com/InnoQueue/Backend/issues/123

**Comment by SMore-Napi** [comment]
Yeah, the merge button has only one available option.

<img width="333" alt="Screenshot 2022-11-14 at 19 54 02" src="https://user-images.githubusercontent.com/49106163/201719090-286ec0d1-a80e-4d53-9f3b-1ef22f11ddb0.png">
